server/src/test/java/org/elasticsearch/search/SearchHitTests.java

```diff
@@ -110,21 +124,34 @@ public static SearchHit createTestItem(boolean withOptionalInnerHits) {
         if (randomBoolean()) {
             hit.explanation(createExplanation(randomIntBetween(0, 5)));
         }
-        if (withOptionalInnerHits) {
+        if (isInnerHit == false && randomBoolean()) {
             int innerHitsSize = randomIntBetween(0, 3);
             if (innerHitsSize > 0) {
                 Map<String, SearchHits> innerHits = new HashMap<>(innerHitsSize);
                 for (int i = 0; i < innerHitsSize; i++) {
-                    innerHits.put(randomAlphaOfLength(5), SearchHitsTests.createTestItem());
+                    innerHits.put(randomAlphaOfLength(5),
+                        SearchHitsTests.createTestItem(xContentType, true));
                 }
                 hit.setInnerHits(innerHits);
             }
         }
-        if (randomBoolean()) {
+        /**
```

Comment:
this should not be a javadoc comment I think. Also s/serialize/serializes  and s/looses/loses .